AppCenterData/AppCenterData/Internal/LocalStore/MSDBDocumentStore.m

```diff
@@ -189,6 +191,108 @@ - (MSDocumentWrapper *)readWithToken:(MSTokenResult *)token documentId:(NSString
                                           fromDeviceCache:YES];
 }
 
+- (MSPaginatedDocuments *)listWithToken:(MSTokenResult *)token
+                              partition:(NSString *)partition
+                           documentType:(Class)documentType
+                            baseOptions:(MSBaseOptions *_Nullable)baseOptions {
+
+  // Final list of documents.
+  NSMutableArray<MSDocumentWrapper *> *localListItems = [NSMutableArray new];
+
+  // Get effective device time to live.
+  NSInteger deviceTimeToLive = baseOptions ? baseOptions.deviceTimeToLive : kMSDataTimeToLiveDefault;
+
+  // Execute the query.
+  NSString *tableName = [MSDBDocumentStore tableNameForPartition:token.partition];
+  NSString *selectionQuery =
+      [NSString stringWithFormat:@"SELECT * FROM \"%@\" WHERE \"%@\" = \"%@\"", tableName, kMSPartitionColumnName, token.partition];
+  NSArray *listResult = [self.dbStorage executeSelectionQuery:selectionQuery];
+
+  // Return an error if no documents were found.
+  if (listResult.count == 0) {
```

Comment:
Since in the iteration on cached documents, first condition checks if the document has expired prior to adding it to the local list ; it would be better to move this condition after the for loop and test the local list.